### PR TITLE
Fix issue where force deadlines weren't respected

### PR DESCRIPTION
**Pull request by dadgar**
The deadline heap was not properly firing for force deadlines. Added unit and integration test.

### CODE REVIEW

File: nomad/drainer/drain_heap.go [hunk 1]
```diff
@@ -96,7 +91,7 @@ func (d *deadlineHeap) watch() {
 			continue
 		}
 
-		if !deadline.Equal(nextDeadline) {
+		if deadline.IsZero() || !deadline.Equal(nextDeadline) {
```
Comment (chelseakomlo) [inline nomad/drainer/drain_heap.go]: Add a comment to explain why either of these cases is valid.

File: nomad/drainer_int_test.go [hunk 2]
```diff
@@ -633,3 +633,123 @@ func TestDrainer_AllTypes_NoDeadline(t *testing.T) {
...
+	}, func(err error) {
+		t.Fatalf("err: %v", err)
+	})
+
+	// Drain the node
+	drainReq := &structs.NodeUpdateDrainRequest{
+		NodeID: n1.ID,
+		DrainStrategy: &structs.DrainStrategy{
+			DrainSpec: structs.DrainSpec{
+				Deadline: 0 * time.Second, // Infinite
```
Comment (chelseakomlo) [inline nomad/drainer_int_test.go]: Would it be valuable to have a test (if there isn't one already) for a non-infinite deadline?

File: nomad/drainer_int_test.go [hunk 3]
```diff
@@ -633,3 +633,123 @@ func TestDrainer_AllTypes_NoDeadline(t *testing.T) {
...
+		DrainStrategy: &structs.DrainStrategy{
+			DrainSpec: structs.DrainSpec{
+				Deadline: -1 * time.Second, // Infinite
+			},
+		},
+		WriteRequest: structs.WriteRequest{Region: "global"},
+	}
+	require.Nil(msgpackrpc.CallWithCodec(codec, "Node.UpdateDrain", drainReq, &drainResp))
+
+	// Make sure the batch job is migrated
```
Comment (chelseakomlo) [inline nomad/drainer_int_test.go]: Should there be a test for when a batch job (or other jobs) cannot be migrated?

Comment (dadgar) [reply nomad/drainer_int_test.go]: From the perspective of the drainer there is no scenario in which it cannot be migrated. You can see that this test only creates one node so there is no possible replacement allocation but we can still assert it got killed.

File: nomad/drainer_int_test.go (hunk 2 above)
Comment (dadgar) [reply nomad/drainer_int_test.go]: Yeah why not!

File: nomad/drainer/drain_heap.go (hunk 1 above)
Comment (dadgar) [reply nomad/drainer/drain_heap.go]: Done.

**Pull request merged**